src/field/legend_cutscene_camera.c

```diff
@@ -0,0 +1,987 @@
+    draw3dTaskData->state = CLEAR_BELL_ANIM_STATE_RISE_BEGIN;
+}
+
+void LegendCutscene_ClearBellShimmer(FieldSystem *fieldSystem, u8 shimmerClearBellOnly) {
+    ClearBellCutsceneCamera *cam = fieldSystem->unk4->legendCutsceneCamera;
+    ClearBellCutscene3dObjectTaskData *draw3dTaskData = (ClearBellCutscene3dObjectTaskData *)Field3dObjectTask_GetData(cam->draw3dTask);
+    if (draw3dTaskData->state != CLEAR_BELL_ANIM_STATE_IDLE_LOOP) {
+        GF_ASSERT(FALSE);
+        TaskManager_Call(fieldSystem->taskman, Task_WaitShimmerEffectAndRestart, NULL);
+        return;
+    }
+    draw3dTaskData->shimmerClearBellOnly = shimmerClearBellOnly;
+    startBellShimmer(fieldSystem);
+}
+
+// Part of an unreferenced code path
+void ov02_02250B30(FieldSystem *fieldSystem) {
+    TaskManager_Call(fieldSystem->taskman, ov02_02251320, NULL);
+}
+
+static void startBellShimmer(FieldSystem *fieldSystem) {
+    ClearBellCutsceneCamera *cam = fieldSystem->unk4->legendCutsceneCamera;
+    ClearBellCutscene3dObjectTaskData *taskData = (ClearBellCutscene3dObjectTaskData *)Field3dObjectTask_GetData(cam->draw3dTask);
+    taskData->state = CLEAR_BELL_ANIM_STATE_SHIMMER_BEGIN;
+}
+
+static BOOL Task_WaitShimmerEffectAndRestart(TaskManager *taskman) {
+    FieldSystem *fieldSystem = TaskManager_GetFieldSystem(taskman);
+    ClearBellCutsceneCamera *cam = fieldSystem->unk4->legendCutsceneCamera;
+    ClearBellCutscene3dObjectTaskData *taskData = (ClearBellCutscene3dObjectTaskData *)Field3dObjectTask_GetData(cam->draw3dTask);
+    if (taskData->state == CLEAR_BELL_ANIM_STATE_IDLE_LOOP) {
+        startBellShimmer(fieldSystem);
+        return TRUE;
+    }
+
+    return FALSE;
+}
+
+void LegendCutscene_LugiaEyeGlimmerEffect(FieldSystem *fieldSystem) {
+    LegendCutsceneLugiaEyeGlimmerTaskData *eyeGlimmer = AllocFromHeapAtEnd(HEAP_ID_4, sizeof(LegendCutsceneLugiaEyeGlimmerTaskData));
+    eyeGlimmer->eyeGlimmerDelayTimer = 0;
+    loadEyeGlimmer3dModel(HEAP_ID_4, fieldSystem, eyeGlimmer);
+    TaskManager_Call(fieldSystem->taskman, Task_LugiaEyeGlimmer, eyeGlimmer);
+}
+
+static void ClearBellCutscene_CreateField3dObjectTask(ClearBellCutsceneCamera *cam) {
+    cam->draw3dTask = Field3dObjectTaskManager_CreateTask(cam->field3dObjectTaskManager, &sField3dObjectTaskTemplate_ClearBellCutscene);
+}
+
+static void Field3dObjectTaskInit_ClearBellCutscene(Field3dObjectTask *task, FieldSystem *fieldSystem, void *taskData) {
+    ClearBellCutscene3dObjectTaskData *kimonoDanceObjData = (ClearBellCutscene3dObjectTaskData *)taskData;
+    ClearBellCutsceneCamera *cam = fieldSystem->unk4->legendCutsceneCamera;
+    kimonoDanceObjData->gameVersion = cam->gameVersion;
+    GF_ExpHeap_FndInitAllocator(&kimonoDanceObjData->allocator, HEAP_ID_4, 0x20);
+    Field3dModel_LoadFromFilesystem(&kimonoDanceObjData->clearBellRisingModel, NARC_demo_legend, NARC_legend_legend_00000018_NSBMD, HEAP_ID_4);
```

Comment:
you can name these legend_00000XX_NSBMD files since it's clear what they are, but I won't block merge over it

Reply:
subject for a future pr, will require separate tinkering with .knarcorder